test/e2e/gateway_api_test.go

```diff
@@ -191,6 +195,107 @@ func testGatewayAPIObjects(t *testing.T) {
 	}
 }
 
+// testGatewayAPIManualDeployment verifies that Istio's manual deployment is not
+// enabled.  When manual deployment is enabled, then Istio allows a gateway to
+// use another gateway's service by specifying that gateway's service in
+// spec.addresses.  This results in behavior similar to gateway listener
+// merging, which we do not want to allow at this time.  Instead, we expect
+// Istio to provision a service for this specific gateway, even if it specifies
+// spec.addresses.
+func testGatewayAPIManualDeployment(t *testing.T) {
+	gatewayClass, err := createGatewayClass("openshift-default", "openshift.io/gateway-controller/v1")
+	if err != nil {
+		t.Fatalf("Failed to create gatewayclass: %v", err)
+	}
+
+	gatewayName := types.NamespacedName{
+		Name:      "manual-deployment",
+		Namespace: "openshift-ingress",
+	}
+	gateway := gatewayapiv1.Gateway{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      gatewayName.Name,
+			Namespace: gatewayName.Namespace,
+		},
+		Spec: gatewayapiv1.GatewaySpec{
+			GatewayClassName: gatewayapiv1.ObjectName(gatewayClass.Name),
+			Addresses: []gatewayapiv1.GatewayAddress{{
+				Type:  ptr.To(gatewayapiv1.HostnameAddressType),
+				Value: "lb.example.com",
```

Comment:
Following up on https://github.com/openshift/cluster-ingress-operator/pull/1204#discussion_r2017371544, shouldn't the spec.Address point to an existing service?  Otherwise, how do we know that the manual deployment didn't just fail because the service didn't exist?

Reply:
> Following up on [#1204 (comment)](https://github.com/openshift/cluster-ingress-operator/pull/1204#discussion_r2017371544), shouldn't the spec.Address point to an existing service? Otherwise, how do we know that the manual deployment didn't just fail because the service didn't exist?

We know that because the gateway was accepted and Istio created a service for it.